clusterctl/clusterdeployer/clusterclient.go

```diff
@@ -171,8 +285,72 @@ func (c *clusterClient) WaitForClusterV1alpha1Ready() error {
 	return waitForClusterResourceReady(c.clientSet)
 }
 
+func (c *clusterClient) waitForClusterDelete() error {
+	return util.PollImmediate(RetryIntervalResourceDelete, TimeoutResourceDelete, func() (bool, error) {
+		glog.V(2).Infof("Waiting for cluster objects to be deleted...")
+		response, err := c.clientSet.ClusterV1alpha1().Clusters(apiv1.NamespaceDefault).List(metav1.ListOptions{})
+		if err != nil {
+			return false, nil
+		}
+		if len(response.Items) > 0 {
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+func (c *clusterClient) waitForMachineDeploymentsDelete() error {
+	return util.PollImmediate(RetryIntervalResourceDelete, TimeoutResourceDelete, func() (bool, error) {
+		glog.V(2).Infof("Waiting for machine deployment objects to be deleted...")
+		response, err := c.clientSet.ClusterV1alpha1().MachineDeployments(apiv1.NamespaceDefault).List(metav1.ListOptions{})
+		if err != nil {
+			return false, nil
+		}
+		if len(response.Items) > 0 {
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+func (c *clusterClient) waitForMachineSetsDelete() error {
+	return util.PollImmediate(RetryIntervalResourceDelete, TimeoutResourceDelete, func() (bool, error) {
+		glog.V(2).Infof("Waiting for machine set objects to be deleted...")
+		response, err := c.clientSet.ClusterV1alpha1().MachineSets(apiv1.NamespaceDefault).List(metav1.ListOptions{})
+		if err != nil {
+			return false, nil
+		}
+		if len(response.Items) > 0 {
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+func (c *clusterClient) waitForMachinesDelete() error {
+	return util.PollImmediate(RetryIntervalResourceDelete, TimeoutResourceDelete, func() (bool, error) {
+		glog.V(2).Infof("Waiting for machine objects to be deleted...")
+		response, err := c.clientSet.ClusterV1alpha1().Machines(apiv1.NamespaceDefault).List(metav1.ListOptions{})
+		if err != nil {
+			return false, nil
```

Comment:
Sorry, I missed this on the first round, shouldn't `err` should be returned here? Same thing in the other wait for delete methods.

Reply:
err should not be returned, from the PollImmediate docs: https://godoc.org/k8s.io/apimachinery/pkg/util/wait#PollImmediate

`PollImmediate tries a condition func until it returns true, an error, or the timeout is reached.`

This means that if err is returned the polling will cease.

Reply:
Actually, nevermind. I see other examples of PollImmediate being used where the error is not returned.

Reply:
Oh you responded before me! Yes, you are right :)